src/simulation/Benchmark.cpp

```diff
@@ -0,0 +1,431 @@
+        {
+        }
+    };
+    return make_unique<BenchmarkExt>(
+        app.getMetrics(), mTxRate,
+        std::unique_ptr<TxSampler>(sampler.release()));
+}
+
+void
+Benchmark::BenchmarkBuilder::prepareBenchmark(
+    Application& app, ShuffleLoadGenerator& sampler) const
+{
+    populateAccounts(app, mAccounts, sampler);
+}
+
+void
+Benchmark::BenchmarkBuilder::populateAccounts(
+    Application& app, size_t size, ShuffleLoadGenerator& sampler) const
+{
+    for (size_t accountsLeft = size, batchSize = size; accountsLeft > 0;
+         accountsLeft -= batchSize)
+    {
+        batchSize = std::min(accountsLeft, MAXIMAL_NUMBER_OF_TXS_PER_LEDGER);
+        auto newAccounts = sampler.createAccounts(batchSize);
+        createAccountsDirectly(app, newAccounts);
+    }
+}
+
+void
+Benchmark::BenchmarkBuilder::setMaxTxSize(LedgerManager& ledger,
+                                          uint32_t maxTxSetSize) const
+{
+    StellarValue sv(ledger.getLastClosedLedgerHeader().hash,
+                    ledger.getLedgerNum(), emptyUpgradeSteps, 0);
+    {
+        LedgerUpgrade up(LEDGER_UPGRADE_MAX_TX_SET_SIZE);
+        up.newMaxTxSetSize() = maxTxSetSize;
+        Value v(xdr::xdr_to_opaque(up));
+        sv.upgrades.emplace_back(v.begin(), v.end());
+    }
+    LedgerCloseData ledgerData = createData(ledger, sv);
+    ledger.closeLedger(ledgerData);
+}
+
+LedgerCloseData
+Benchmark::BenchmarkBuilder::createData(LedgerManager& ledger,
+                                        StellarValue& value) const
+{
+    auto ledgerNum = ledger.getLedgerNum();
+    TxSetFramePtr txSet =
+        std::make_shared<TxSetFrame>(ledger.getLastClosedLedgerHeader().hash);
+    value.txSetHash = txSet->getContentsHash();
+    return LedgerCloseData{ledgerNum, txSet, value};
+}
+
+void
+Benchmark::BenchmarkBuilder::createAccountsDirectly(
+    Application& app,
+    std::vector<LoadGenerator::AccountInfoPtr>& createdAccounts) const
+{
+    soci::transaction sqlTx(app.getDatabase().getSession());
+
+    auto ledger = app.getLedgerManager().getLedgerNum();
+
+    int64_t balanceDiff = 0;
+    std::vector<LedgerEntry> live;
+    std::transform(
+        createdAccounts.begin(), createdAccounts.end(),
+        std::back_inserter(live),
+        [&app, &balanceDiff](LoadGenerator::AccountInfoPtr const& account) {
+            AccountFrame aFrame = account->createDirectly(app);
+            balanceDiff += aFrame.getBalance();
+            return aFrame.mEntry;
+        });
+
+    SecretKey skey = SecretKey::fromSeed(app.getNetworkID());
+    AccountFrame::pointer masterAccount =
+        AccountFrame::loadAccount(skey.getPublicKey(), app.getDatabase());
+    LedgerDelta delta(app.getLedgerManager().getCurrentLedgerHeader(),
+                      app.getDatabase());
+    masterAccount->addBalance(-balanceDiff);
+    masterAccount->touch(ledger);
+    masterAccount->storeChange(delta, app.getDatabase());
+
+    sqlTx.commit();
```

Comment:
not 100% sure, but I think that trying to create millions of account in a single tx like this will be extremely slow